src/app/GitUI/CommandsDialogs/WorktreeDialog/FormCreateWorktree.cs

```diff
@@ -3,176 +3,202 @@
+                    });
 
-                    ValidateWorktreeOptions();
-                });
+                await this.SwitchToMainThreadAsync();
+                if (comboBoxBranches.Items.Count == 0)
+                {
+                    radioButtonCreateNewBranch.Checked = true;
+                    radioButtonCheckoutExistingBranch.Enabled = false;
+                }
+                else
+                {
+                    radioButtonCheckoutExistingBranch.Checked = true;
+                }
 
-                return Task.CompletedTask;
-            }
+                ValidateWorktreeOptions();
+            });
+
+            return Task.CompletedTask;
         }
+    }
 
-        protected override void Dispose(bool disposing)
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
         {
-            if (disposing)
-            {
-                _branchesLoader.Dispose();
-
-                components?.Dispose();
-            }
+            _branchesLoader.Dispose();
 
-            base.Dispose(disposing);
+            components?.Dispose();
         }
 
-        private void comboBoxBranches_KeyUp(object sender, KeyEventArgs e)
+        base.Dispose(disposing);
+    }
+
+    private void comboBoxBranches_KeyUp(object sender, KeyEventArgs e)
+    {
+        if (e.KeyCode == Keys.Enter)
         {
-            if (e.KeyCode == Keys.Enter)
-            {
-                CreateWorktree();
-            }
+            CreateWorktree();
         }
+    }
 
-        private void createWorktreeButton_Click(object sender, EventArgs e)
+    private void createWorktreeButton_Click(object sender, EventArgs e)
+    {
+        CreateWorktree();
+    }
+
+    private void CreateWorktree()
+    {
+        string relativePath = Path.GetRelativePath(
+            Module.WorkingDir, WorktreeDirectory).ToPosixPath().Quote();
+        string newBranchOption =
+            radioButtonCreateNewBranch.Checked ?
+            $"-b {textBoxNewBranchName.Text}" :
+            comboBoxBranches.SelectedItem is not null ? ((GitRef)comboBoxBranches.SelectedItem).Name : null;
+
+        DialogResult = UICommands.StartGitCommandProcessDialog(this, CreateWorktreeCommand(Module, relativePath, newBranchOption)) ? DialogResult.OK : DialogResult.None;
+    }
+
+    private GitArgumentBuilder CreateWorktreeCommand(IGitModule module, string relativePath, string newBranchOption)
+    {
+        // https://git-scm.com/docs/git-worktree
+
+        // Get the default value, set if unset in config.
+        // Similar in DiffHighlightService.
+        string command = "worktree";
+        GitCommandConfiguration commandConfiguration = new();
+        IReadOnlyList<GitConfigItem> items = GitCommandConfiguration.Default.Get(command);
+        foreach (GitConfigItem cfg in items)
         {
-            CreateWorktree();
+            commandConfiguration.Add(cfg, command);
         }
```

Comment:
... and in GrepHighLightService.
So, it seems to be the time for creating `GitCommandConfiguration.Clone` and `GitCommandConfiguration.SetIfUnsetInGit`.

Reply:
ok to take that separately?